sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -1548,6 +1678,166 @@ different from the one with which the dynamic_parameter was created.
 
 |===
 
+=== Event Class Modifications [[event-class-modifications]]
+
+[source, c++]
+----
+namespace sycl::info {
+  enum class ext_oneapi_event_t{
+    normal,
+    graph_limited,
+    graph_external
+  };
+} // namespace sycl::info
+----
+
+:event-class: https://www.khronos.org/registry/SYCL/specs/sycl-2020/html/sycl-2020.html#sec:interface.event
+
+==== Event Type Query
+
+This extension adds the following new information descriptor to the event class
+for use with `event::get_info<>()` which can be used to query the type of a SYCL
+event.
+
+Table {counter: tableNumber}. Additions to the `sycl::event` class information descriptors.
+[%header,cols="2a,a,a"]
+|===
+|Event Descriptor
+|Return Type
+|Description
+
+|info::event::ext_oneapi_event_type
+|info::ext_oneapi_event_t
+|Returns the type of this SYCL event
+|===
+
+Events of type `ext_oneapi_event_t::normal` are normal SYCL events as described
+in the SYCL specification. `ext_oneapi_event_t::graph_limited` and
+`ext_oneapi_event_t::graph_external` events are new as part of this extension,
+and are differentiated from normal SYCL events in how they can be used.
+
+==== Graph-Limited Events
+
+Events associated with a graph node (those returned from recorded queue
+submissions) cannot be used as normal SYCL events outside of their associated
+graph, these are known as "graph-limited" events. They can only be used as
+parameters to `handler::depends_on()`, or as dependent events for queue
+shortcuts like `queue::parallel_for()` for other submissions recorded to
+the same modifiable `command_graph`.
+
+The following limitations apply to graph-limited events:
+
+- Calling `event::get_info<info::event::command_execution_status>()` or
+`event::get_profiling_info()` will throw synchronously with error code
+`invalid`.
+
+- A host-side wait on the event will throw synchronously with error
+code `invalid`.
+
+- Using the event outside of the recording scope will throw synchronously with
+error code `invalid`.
+
+==== External Events
```

Comment:
We might need to revisit our definition of edges in the section on https://github.com/reble/llvm/blob/a3cb220cf85b7fc04830885e53c187c6785dd77e/sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc#72-sycl-graph-terminology for record & replay and explicit to take this into account.

Reply:
I've reworded those sections a little bit to point to the event changes for more information.